### PR TITLE
All merchants records actions

### PR DESCRIPTION
**Pull request by wlffann**
This creates the merchants end points for retrieving records.

Still needs find_all endpoints.
Some tests not passing. Still figuring out date searches.

### CODE REVIEW

File: spec/requests/api/v1/merchants_spec.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    found_merchants = JSON.parse(response.body)
+
+    expect(response).to be_success
+    expect(found_merchants).to be_a(Array)
+    expect(found_merchants.first["id"]).to eq(Merchant.first.id)
+    expect(found_merchants.first["name"]).to eq(Merchant.first.name)
+  end
+
+  it "can find all records matching name" do
+    create_list(:merchant, 3)
```
Comment (drod1000) [inline spec/requests/api/v1/merchants_spec.rb:92]: @wlffann Do you think we should create a separate merchant with a different name that wouldn't get picked up by the search? That way we can confirm that it's filtering out those that don't match and not just returning all of the merchants every time?

**Review by drod1000 — COMMENTED** [review]
@wlffann My only concern with the find all tests is that the number of merchants being returned is the same as the total number of merchants in the database. Your code looks good so I'm sure it's working properly but I think it might be good to have our tests show that the search is filtering out those that don't match. Other than that I think everything looks great!

File: spec/requests/api/v1/merchants_spec.rb (hunk 1 above)
Comment (wlffann) [reply spec/requests/api/v1/merchants_spec.rb:92]: Sure! I think that's a good suggestion. I'll update those tests and factories now.

**Comment by wlffann** [comment]
@case-eee

File: app/controllers/api/v1/random_merchants_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::RandomMerchantsController < ApplicationController
+
+  def show
+    render json: Merchant.random
```
Comment (case-eee) [inline app/controllers/api/v1/random_merchants_controller.rb:4]: 👍

File: app/models/merchant.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Merchant < ApplicationRecord
+
+  def self.random
```
Comment (case-eee) [inline app/models/merchant.rb:3]: 👍

File: spec/requests/api/v1/merchants_spec.rb (hunk 1 above)
Comment (case-eee) [reply spec/requests/api/v1/merchants_spec.rb:92]: i like that suggestion @drod1000 🎉

**Review by case-eee — COMMENTED** [review]
looks good to me @wlffann 👍 

i've also seen nesting your controllers further - so for example, you have a `merchants` folder under `api/vi/` then your controller is `Api::V1::Merchants::RandomController`

does that make sense?!

**Pull request merged**